zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/UserController.java

```diff
@@ -46,4 +53,28 @@ public CompletableFuture<ResponseEntity<Object>> createUser(
                     userWithPasswordDto.getEmail(),
                     passwordEncoder.encode(userWithPasswordDto.getPassword())));
   }
+
+  @PostMapping(
```

Comment:
⚠️ If we add a search endpoint to a REST controller annotated with `CamundaRestController`, the endpoint will be available in a cluster by default. 

For search/query endpoints, we currently don't do this, see our [public documentation](https://docs.camunda.io/docs/next/apis-tools/camunda-api-rest/camunda-api-rest-overview/#query-api) for the reasoning.

For all other search endpoints, we currently create specific controllers annotated with `CamundaRestQueryController`. Those are not enabled in a cluster by default.

It could be fine here, I just want to make you aware of this. If every authenticated user should be allowed to query all users, i.e. there are no resource permissions people can configure for this resource, we can probably keep it as is.

Reply:
Good point @tmetzke, I wasn't aware of this and as I understand we shouldn't allow all authenticate user to query all users. I'm going to move this endpoint to a separate controller.